### PR TITLE
feat: add user command, option to disable trimming whitespace

### PR DESCRIPTION
**Pull request by ofirgall**

### CODE REVIEW

File: lua/retrail/init.lua [hunk 1]
```diff
@@ -21,14 +21,16 @@ function M.setup(opts)
...
   })
-  vim.api.nvim_create_autocmd("BufWritePre", {
-    group = augroup,
-    callback = function()
-      if M:enabled() then
-        M:trim()
-      end
-    end,
-  })
+  if M.config.auto then
```
Comment (kaplanz) [inline lua/retrail/init.lua]: Should this say `M.config.trim.auto`?

File: lua/retrail/init.lua [hunk 2]
```diff
@@ -51,6 +53,10 @@ function M.setup(opts)
     desc = "Toggle for the current buffer",
     nargs = 0,
   })
+  vim.api.nvim_create_user_command('TrimWhitespace', function()
```
Comment (kaplanz) [inline lua/retrail/init.lua]: I like the idea of setting a user command, but I feel we should keep this using the format `Retrail*`. Do you like the idea of `RetrailTrim`, or `RetrailTrimWhitespace`?

**Review by kaplanz — CHANGES_REQUESTED** [review]

**Comment by kaplanz** [comment]
Thanks for the contribution, and sorry for the delay in my review! I made some suggestions, but otherwise it looks great!

File: lua/retrail/init.lua (hunk 1 above)
Comment (ofirgall) [reply lua/retrail/init.lua]: Yes.

**Comment by ofirgall** [comment]
Thank you for writing this plugin it took me 3 mins to add the functionally I missed, it's well written.

**Comment by kaplanz** [comment]
Merged, thanks!

**Pull request merged**